File: 443-string-compression.cs
public class Solution {
    public int Compress(char[] chars) {
        var length = chars.Length;
        int slow = 0, fast = 0, cur = 0;
        while (fast <= length) {
            if (fast == length || chars[fast] != chars[slow]) {
                chars[cur++] = chars[slow];
                if (fast - slow >= 2)
                    foreach (var ch in (fast - slow).ToString())
                        chars[cur++] = ch;
                slow = fast;
            }

            fast++;
        }

        return chars.Take(cur).Count();
    }
}

File: 443-string-compression.go
package main

import (
	"strconv"
)

func compress(chars []byte) int {
	n := len(chars)
	slow, fast, cur := 0, 0, 0
	for fast <= n {
		if fast == n || chars[fast] != chars[slow] {
			chars[cur] = chars[slow]
			cur++
			if (fast - slow) >= 2 {
				for _, ch := range strconv.Itoa(fast - slow) {
					chars[cur] = byte(ch)
					cur++
				}
			}
			slow = fast
		}
		fast++
	}
	return cur
}

func main() {
	compress([]byte{'a', 'a', 'b', 'b', 'c', 'c', 'c'})
}
